casual/casual-caller/src/test/groovy/se/laz/casual/connection/caller/TransactionPoolMapperTest.groovy

```diff
@@ -0,0 +1,61 @@
+package se.laz.casual.connection.caller
+
+
+import spock.lang.Shared
+import spock.lang.Specification
+
+import javax.transaction.Status
+import javax.transaction.Transaction
+
+class TransactionPoolMapperTest extends Specification {
+    @Shared
+    TransactionManagerImpl transactionManager
+
+    def setup()
+    {
+        TransactionPoolMapper.resetForTest()
+        TransactionPoolMapper.instance().setActiveForTest(true)
+        TransactionPoolMapper.instance().setTransactionManager(transactionManager = new TransactionManagerImpl())
+    }
+
+    def cleanupSpec()
+    {
+        TransactionPoolMapper.resetForTest()
+    }
+
+    def "Test default config state, should be disabled/inactive"() {
+        setup:
+        TransactionPoolMapper.resetForTest()
+
+        expect:
+        !TransactionPoolMapper.instance().isPoolMappingActive()
+    }
+
+    def "Test with single transaction"() {
+        setup:
+        Transaction transaction = new TransactionImpl(Status.STATUS_ACTIVE)
+        transactionManager.setCurrentTransaction(transaction)
+
+        String actualPoolName = "hello, world!"
+        TransactionPoolMapper.instance().setPoolNameForCurrentTransaction(actualPoolName)
+
+        expect:
+        TransactionPoolMapper.instance().getPoolNameForCurrentTransaction() == actualPoolName
+        TransactionPoolMapper.instance().getNumberOfTrackedTransactions() == 1
+    }
+
+    def "Test with many transactions"()
+    {
+        setup:
+        int transactions = 5
+
+        for (int i = 0; i < transactions; i++)
+        {
+            transactionManager.setCurrentTransaction(new TransactionImpl(Status.STATUS_ACTIVE))
+            TransactionPoolMapper.instance().setPoolNameForCurrentTransaction("hello"+i)
```

Comment:
Should always be the same for the same transaction, but in this case we are using different transactions to add different pools just to see that we get a correct number of tracked transactions in the end. For the purpose of the test using different pool names for each transaction has no real bearing though since the poolmapper does not care if 5 different transactions stick to the same or different pools

Reply:
Sticky with the same transaction but different pools would be an error though currently, no?

Reply:
Normally we should never set a sticky for a transaction where it is already set. Technically you could do it and it would result in the stickied pool being changed to the last pool that was set for the given transaction. If the test in question used the same transaction five times it would be expected in the end that we would be tracking one stickied transaction instead of five with how it is implemented

Reply:
I'm reading this with the eyes of somebody who does not know how it is supposed to work.
That is, that we only ever sticky a transaction to one pool only - anything else is not ok.
This might confuse that?

Reply:
I made the test clearer and added a new test to verify a more strict behavior for when multiple attempts are made to set the sticky for the same transaction